### ISSUE TITLE
POST/PUT response handling expects full object is returned in repsonse

### ISSUE THREAD

**Issue by mallison** [Kvoti/redux-rest#3]
Again, this is specific to Django Rest Framework's defaults. API's may only return the object ID on successful create or update. We need to handle that case.

**Comment by bhoomit** [comment]
On similar note my list API returns custom object containing multiple lists. Right now there's no way to accept such response. Is it intentional?

**Comment by mallison** [comment]
It's intentional in the sense that I only did enough to get things working with my backend. Adding ```setCSRF``` (now ```SetHeader```!) was a start at generalising things.

Do you have an example of returning multiple lists? We could allow further customisation if there's a good use case.

**Comment by bhoomit** [comment]
There's a home screen where I return multiple type of cards(data). E.g. Region, City, Hotel, Staff Picks, etc. The response is something like 

`{
"regions": [{},{},{}],
"cities": [{},{}],
"hotels": [{},{}]
}`

**Comment by mallison** [comment]
I think, strictly speaking, that isn't quite RESTful but, for pragmatic reasons, most APIs aren't. I could imagine something like:

```js
const API = {
    '/api/': {
         "regions": types.List(...)
         "cities": types.List(...)
    }
};
```

but that's not really RESTful and outside the scope of this library I think. The best thing is probably to make sure you can easily use redux-rest, for 'regular' endpoints, along side your own code for special cases.

**Comment by bhoomit** [comment]
I totally understand that. But on mobile devices for India like demographics we try to minimize number of API calls.

**Comment by mallison** [comment]
Yes, I understand that (hence my pragmatism point). It's also why
Relay/GraphQL is the probably the way forward! It's something I (we) can
thing about. Did my pseudo-code make sense? Something like that might work.
It would add a layer of complexity but maybe not too much.

**Comment by bhoomit** [comment]
Yes, It does makes sense. I didn't know about Relay/GraphQL. So went ahead and read about it. And yes it does seem like the way forward. Unfortunately, right now its hard for us to move to that as there are apps which are already using the APIs. Anyway thanks for the tip.

**Comment by mallison** [comment]
No problem. I think REST APIs will be around for a while which is why I
started redux-rest. I'm open to exploring aggregate endpoints like your
example. As you say, it must be pretty common for low-bandwidth areas.

**Comment by bhoomit** [comment]
For now what I did in my fork is this:

```
In ItemReducer

} else if (action.type === this.actionTypes.list_success) {
      if (Array.isArray(action.payload)){
        return [...action.payload];  
      }
      return action.payload;
}
```

**Comment by mallison** [comment]
Ok, that's fine for retrieving items. I guess creating, updating items
doesn't work?

**Comment by bhoomit** [comment]
This was just a quick fix. I didn't test it with other operations. Will check when I get there. Once I'm sure about it, I will send a PR.

**Comment by mallison** [comment]
Great, thanks.
